File: src/Java/BehavioralPatterns/Mediator/CommunicationHub/src/Mediators/DirectMediator.java
package Mediators;

import Participants.Participant;

import java.util.List;

public class DirectMediator implements IMediator {
    @Override
    public void notify(Participant sender, Object senderArgs) {
        if (!(senderArgs instanceof List<?>)) {
            return;
        }

        List<?> argsList = (List<?>) senderArgs;

        if (argsList.size() < 2 || !(argsList.get(0) instanceof Participant)) {
            return;
        }

        Participant receiver = (Participant) argsList.get(0);
        receiver.receive(sender, argsList.get(1));
    }
}

File: src/PHP/BehavioralPatterns/Mediator/CommunicationHub/Mediators/DirectMediator.php
<?php

namespace BehavioralPatterns\Mediator\CommunicationHub\Mediators;

use BehavioralPatterns\Mediator\CommunicationHub\Participants\Participant;

class DirectMediator implements IMediator
{
    public function notify($sender, $senderArgs): void
    {
        if (!is_array($senderArgs)) {
            return;
        }

        $receiver = $senderArgs[0];
        $args = $senderArgs[1];

        if (!($receiver instanceof Participant)) {
            return;
        }

        $receiver->receive($sender, $args);
    }
}